modules/core/elb.tf

```diff
@@ -110,8 +110,9 @@ resource "aws_lb_listener_rule" "nomad_server" {
   }
 
   condition {
-    field  = "host-header"
-    values = [var.nomad_api_domain]
+    host-header {
+      values = [var.nomad_api_domain]
+    }
```

Comment:
For this change, see: 
Documentation for new syntax: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_listener_rule
Issue and when it was merged: 
https://github.com/terraform-providers/terraform-provider-aws/issues/8126